internal/plans/planfile/tfplan.go

```diff
@@ -470,13 +478,16 @@ func changeFromTfplan(rawChange *planproto.Change) (*plans.ChangeSrc, error) {
 }
 
 func valueFromTfplan(rawV *planproto.DynamicValue) (plans.DynamicValue, error) {
-	if len(rawV.Msgpack) == 0 { // len(0) because that's the default value for a "bytes" in protobuf
+	if valueIsEmpty(rawV) {
 		return nil, fmt.Errorf("dynamic value does not have msgpack serialization")
 	}
-
 	return plans.DynamicValue(rawV.Msgpack), nil
 }
 
+func valueIsEmpty(rawV *planproto.DynamicValue) bool {
+	return rawV == nil || len(rawV.Msgpack) == 0 // len(0) because that's the default value for a "bytes" in protobuf
```

Comment:
So if I understand correctly, we are determining plan.EphemeralVariables from the null/emptyness of a Dynamic value.

I'm wondering if we want to store plan.EphemeralValues directly in the proto file instead of re-deriving it later on.  I could see this causing some weird bugs down the line?

Reply:
Since the concept is new, as long as those are wrote and read correctly from the plan, there should be no issue.
The only thing that I am not sure about is how the proto changes will impact downgrades, in case are needed 🤔

Reply:
Plan files are locked to specific versions and thus can not be upgraded/downgraded.

IMO it's safer both now and when new features are introduced to explicilty store this information in the proto representation.